src/uwtools/drivers/forecast.py

```diff
@@ -121,15 +129,92 @@ def requirements(self) -> None:
         ???
         """
 
-    def resources(self) -> None:
+    def resources(self, platform: dict) -> Mapping:
         """
-        ???
+        Set up the formatting for the scheduler batch script.
         """
-
-    def run(self):
+        # Add required fields to platform.
+        # Currently supporting only slurm and none.
+        slurmqueues = ["hera", "jet", "orion", "stampede"]
+        if platform["queue"] is None and platform["machine"] in slurmqueues:
+            platform["queue"] = "slurm"
+        else:
+            platform["queue"] = "none"
+
+        return {
+            "account": platform["account"],
+            "nodes": 1,
+            "queue": platform["queue"],
+            "scheduler": platform["scheduler"],
+            "tasks_per_node": 1,
+            "walltime": "00:01:00",
+        }
+
+    def run(self, log: Optional[Logger] = None) -> None:
         """
         Runs FV3.
         """
+        # Set up logging.
+        if log is None:
+            name = f"{inspect.stack()[0][3]}"
+            log = cli_helpers.setup_logging(
+                log_file="/dev/stdout",
+                log_name=name,
+                quiet=False,
+                verbose=False,
+            )
+        # Read in the config file.
+        forecast_config = config.YAMLConfig(self._config_file)
+
+        # Define forecast model.
+        forecast_model = forecast_config["forecast"]["EXTRN_MDL_NAME"]
+        machine = forecast_config["platform"]["machine"]
+
+        # Prepare directories.
+        run_directory = forecast_config["forecast"]["FCSTDIR"]
```

Comment:
We likely want each driver to have the same entry that is something like `run_directory` or `work_dir`. I'm open to others, but it's already in its own namespace, so let's make it consistent and generic.

We also likely want to pare down the scope of what the driver needs to just be in the `forecast` section just after it loads an experiment YAML. That would mean we don't have to repeat asking for the `forecast` section request when we look for config settings the driver needs.